cmd/tsdb/main.go

```diff
@@ -33,35 +31,30 @@ import (
 	"github.com/prometheus/prometheus/pkg/textparse"
 	"github.com/prometheus/tsdb"
 	"github.com/prometheus/tsdb/labels"
-	"github.com/spf13/cobra"
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
 )
 
-func main() {
-	// Start HTTP server for pprof endpoint.
-	go http.ListenAndServe(":9999", nil)
-
-	root := &cobra.Command{
-		Use:   "tsdb",
-		Short: "CLI tool for tsdb",
-	}
-
-	root.AddCommand(
-		NewBenchCommand(),
-	)
+var (
+	cli = kingpin.New(filepath.Base(os.Args[0]), "CLI tool for tsdb")
```

Comment:
Can you move this into main() itself? You could also make benchWriteFile a member of the `writeBenchmark` struct and that way you don't need the global variable.